modules/data.land/R/ic_process.R

```diff
@@ -49,15 +50,25 @@ ic_process <- function(settings, input, dir, overwrite = FALSE){
   con <- PEcAn.DB::db.open(dbparms$bety)
   on.exit(PEcAn.DB::db.close(con), add = TRUE)
   
-  #grab site lat and lon info
-  latlon <- PEcAn.DB::query.site(site$id, con = con)[c("lat", "lon")] 
   # setup site database number, lat, lon and name and copy for format.vars if new input
-  new.site <- data.frame(id = as.numeric(site$id),
-                         lat = latlon$lat,
-                         lon = latlon$lon)
+  latlon <- NULL
+  if(is.null(site$lat) | is.null(site$lon)) {
+    site.info <- PEcAn.DB::get.new.site(site, con=con, latlon = latlon)
 
-  new.site$name <- settings$run$site$name
+    # extract new.site and str_ns from site.info
+    new.site <- site.info$new.site
+    str_ns <- site.info$str_ns
+  } else {
+    latlon <- list(lon = site$lat, lon=site$lon)
+    new.site <- data.frame(
+      id = as.numeric(site$id),
+      lat = site$lat,
+      lon = site$lon
+    )
```

Comment:
I'm not following why this if/else exists. Doesn't get.new.site already support this `else` case? And if not, shouldn't it?